rlmolecule/hallway/hallway_problem.py

```diff
@@ -0,0 +1,177 @@
+            if new_checkpoint != self._checkpoint:
+                self._checkpoint = new_checkpoint
+                status = self.policy_model.load_weights(self._checkpoint)
+                status.assert_existing_objects_matched()
+                logger.info(f'Loaded checkpoint {self._checkpoint}')
+            elif new_checkpoint == self._checkpoint:
+                logger.info(f'Skipping already loaded {self._checkpoint}')
+            else:
+                logger.info('No checkpoint found')
+
+    def _get_network_inputs(self, state: HallwayState) -> Dict:
+        return {"position": [state.position]}
+
+    def _get_batched_network_inputs(self, parent: AlphaZeroVertex) -> Dict:
+        """
+        :return the given nodes policy inputs, concatenated together with the
+        inputs of its successor nodes. Used as the inputs for the policy neural
+        network
+        """
+
+        policy_inputs = [self._get_network_inputs(vertex.state)
+                         for vertex in itertools.chain((parent,), parent.children)]
+        return {key: pad_sequences([elem[key] for elem in policy_inputs], padding='post')
+                for key in policy_inputs[0].keys()}
+
+    def get_value_and_policy(self, parent: AlphaZeroVertex) -> (float, {AlphaZeroVertex: float}):
+
+        values, prior_logits = self.policy_evaluator(self._get_batched_network_inputs(parent))
+
+        # inputs to policy network
+        priors = tf.nn.softmax(prior_logits[1:]).numpy().flatten()
+
+        # Update child nodes with predicted prior_logits
+        children_priors = {vertex: prior for vertex, prior in zip(parent.children, priors)}
+        value = float(tf.nn.sigmoid(values[0]))
+
+        return value, children_priors
+
+
+    def _get_network_inputs_from_serialized_parent(
+            self,
+            serialized_parent: tf.Tensor) -> {}:
+
+        parent = HallwayState.deserialize(serialized_parent.numpy().decode())
+
+        policy_inputs = [self._get_network_inputs(parent)
+                         for state in itertools.chain((parent,), parent.get_next_actions())]
+
+        policy_inputs = {key: pad_sequences([elem[key] for elem in policy_inputs], padding='post')
+                         for key in policy_inputs[0].keys()}
+
+        return policy_inputs['position']
+
+
+    def _create_dataset(self) -> tf.data.Dataset:
+        """
+        Creates a tensorflow dataset pipeline to batch game positions from the replay buffer into
+
+        :param problem:
+        :return:
+        """
+
+        def get_policy_inputs_tf(parent, reward, visit_probabilities,
+                                 problem: HallwayAlphaZeroProblem) -> {}:
+            position = tf.py_function(
+                problem._get_network_inputs_from_serialized_parent, inp=[parent],
+                Tout=[tf.int64])
+            print(position)
+            position.set_shape([None, None])
```

Comment:
is this where you're getting the error? I would think position would be a [None, 1] or [None] shaped vector here?

Yeah, looks like you use it as [None, 1] later as a neural network input. You might want to make sure position coming out of `_get_network_inputs_from_serialized_parent` is a 2D numpy array

Reply:
makes sense, yeah i was working backwards through this.  i had [None, 1] at one point but this also failed, probably for the reason you mentioned